src/rulesets/Base/array.jl

```diff
@@ -262,19 +262,58 @@ function rrule(::typeof(hvcat), rows, values::Union{AbstractArray, Number}...)
 end
 
 #####
-##### `fill`
+##### `reverse`
 #####
 
-function rrule(::typeof(fill), value::Any, dims::Tuple{Vararg{Int}})
-    function fill_pullback(Ȳ)
-        return (NoTangent(), sum(Ȳ), NoTangent())
-    end
-    return fill(value, dims), fill_pullback
+# 1-dim case allows start/stop, N-dim case takes dims keyword, whose defaults changed in Julia 1.6.
+
+function frule((_, xdot, _, _), ::typeof(reverse), x::AbstractVector, start::Integer, stop::Integer)
+    return reverse(x, start, stop), reverse(xdot, start, stop)
+end
+function frule((_, xdot), ::typeof(reverse), x::AbstractArray; dims=VERSION >= v"1.6" ? Colon() : 1)
+    return reverse(x; dims=dims), reverse(xdot; dims=dims)
+end
+# More compact `frule` with splats, should work but doesn't?
+# function frule((_, xdot, _, _), ::typeof(reverse), x::AbstractArray, args...; kw...)
+#     return reverse(x, args...; kw...), reverse(xdot, args...; kw...)
+# end
```

Comment:
```suggestion
# function frule((_, xdot, _argsdots...), ::typeof(reverse), x::AbstractArray, args...; kw...)
#     return reverse(x, args...; kw...), reverse(xdot, args...; kw...)
# end
```
Oh, I see, I think it needs the splatting of the tangents also

Reply:
Oh right, well spotted, thanks!

Reply:
In fact it seems `frule((_, xdot), ::typeof(reverse), x::AbstractArray, args...; kw...)` is enough --- that's like what `fill` already had.